src/components/pincer.rs

```diff
@@ -13,17 +13,33 @@ impl Default for PincerState {
     }
 }
 
+/// Pincer AI is a simple state machine. Pincer either patrols its designated area or
+/// attacks its target.
+#[derive(Eq, Hash, PartialEq, Clone, Copy)]
+pub enum PincerAi {
```

Comment:
Can we merge this enum with the PincerState enum? I think we can have a single enum like the following:
```
pub enum PincerState {
    Dying,
    Idling,
    Patrolling,
    Attacking { target: Entity },
}
```
Idling and Dying are not being used anywhere currently I guess. So, may be we can even get rid of those two.

Reply:
Looking at marine, its State is tightly correlated with animations. I sort of assumed the same would be true for pincer. Since AI should in my opinion be independent of animations, I decided to make a separate enum. To illustrate, image a slightly more advanced AI that pauses to look for the player, chases the player, perhaps even jumps over hurdles to get to the player. These are different animations all utilised within the same AI state. 

Of course, currently, PincerState is not used at all, so we might as well remove it altogether.